src/components/ProfilePage.jsx

```diff
@@ -112,11 +125,37 @@ export class ProfilePage extends React.Component {
           </Row>
           <Row>
             <Col xs={{ order: 2 }} md={{ size: 4, order: 1 }} lg={3} className="mt-md-4">
-              <Name formId="name" {...commonFormProps} />
-              <Country formId="country" {...commonFormProps} />
-              <PreferredLanguage formId="languageProficiencies" {...commonFormProps} />
-              <Education formId="education" {...commonFormProps} />
-              <SocialLinks formId="socialLinks" {...commonFormProps} />
+              <Name
+                name={name}
+                visibilityName={visibilityName}
+                formId="name"
+                {...commonFormProps}
+              />
+              <Country
+                country={country}
+                visibilityCountry={visibilityCountry}
+                formId="country"
+                {...commonFormProps}
+              />
+              <PreferredLanguage
+                languageProficiencies={languageProficiencies}
+                visibilityLanguageProficiencies={visibilityLanguageProficiencies}
+                formId="languageProficiencies"
+                {...commonFormProps}
+              />
+              <Education
+                education={education}
+                visibilityEducation={visibilityEducation}
+                formId="education"
+                {...commonFormProps}
+              />
+              <SocialLinks
+                socialLinks={socialLinks}
+                draftSocialLinksByPlatform={draftSocialLinksByPlatform}
```

Comment:
The other forms get their drafts straight from the store. Is there a reason to pass socialLinks in from the parent?

Reply:
Oh nvm, the others are different than I first thought.

Reply:
They all have their values passed in from ProfilePage.jsx.  The selector for ProfilePage.jsx determines if it should use the draft value (if it exists) or the committed value.  SocialLinks gets it's drafts/committed values passed in on line 153 there - `socialLinks={socialLinks}`.

In addition to that, SocialLinks.jsx needs something extra, which is this drafts lookup.  When editing social links, it wants to push a complete drafts object back to the reducer _in the same order_ as before.  `draftSocialLinksByPlatform` is just a lookup of the links by their platform name so that we can easily reconstruct the drafts data onChange in the same order it was originally in - you can see where it's used in the file itself.